library/src/main/scala/com/monsanto/arch/kamon/prometheus/PrometheusExtension.scala

```diff
@@ -43,6 +50,13 @@ class PrometheusExtension(system: ExtendedActorSystem) extends Kamon.Extension {
   /** The Spray endpoint. */
   val route: Route = endpoint.route
 
+  /** For internal use (healthchecks, other routes, etc.) */
+  def requestSnapshot() = Option(snapshot.get())
+
+  def flushBuffer() = {
+    buffer ! FlushBuffer
+  }
+
```

Comment:
I don't have any current use for _flushBuffer_ except for testing. I could make it package private if you prefer it.

As for _ requestSnapshot_ the reason for it is that my application already has a http server implementation and it's own collections metrics stacks (prometheus, codahale, etc.). Using _requestSnapshot_ made it easier to converge all outputs of all implementations.